src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java

```diff
@@ -1,29 +1,192 @@
 package ch.heigvd.pro.b04.questions;
 
+import ch.heigvd.pro.b04.auth.exceptions.WrongCredentialsException;
+import ch.heigvd.pro.b04.error.exceptions.ResourceNotFoundException;
+import ch.heigvd.pro.b04.moderators.Moderator;
+import ch.heigvd.pro.b04.moderators.ModeratorRepository;
+import ch.heigvd.pro.b04.participants.Participant;
+import ch.heigvd.pro.b04.participants.ParticipantRepository;
+import ch.heigvd.pro.b04.polls.ServerPoll;
+import ch.heigvd.pro.b04.polls.ServerPollIdentifier;
+import ch.heigvd.pro.b04.polls.ServerPollRepository;
 import java.util.List;
+import java.util.Optional;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class QuestionController {
 
   @Autowired
-  private QuestionRepository repository;
+  private final QuestionRepository repository;
+  @Autowired
+  private ServerPollRepository pollRepository;
+  @Autowired
+  private ParticipantRepository participantRepository;
+  @Autowired
+  private ModeratorRepository moderatorRepository;
```

Comment:
I think `@Autowired` is deprecated in Spring, and that we should simply add the fields in the constructor parameters and populate them in the constructor.